python/ray/tune/trial.py

```diff
@@ -111,16 +126,17 @@ def __init__(self,
             resources
             or self._get_trainable_cls().default_resource_request(self.config))
         self.stopping_criterion = stopping_criterion or {}
-        self.checkpoint_freq = checkpoint_freq
         self.upload_dir = upload_dir
         self.verbose = True
         self.max_failures = max_failures
 
         # Local trial state that is updated during the run
         self.last_result = None
-        self._checkpoint_path = restore_path
-        self._checkpoint_obj = None
-        self.runner = None
```

Comment:
`runner` is a platform specific variable and must be bounded to `Trial` instance. Currently it's been set outside the Trial, which may break the integrity of Trial design. Maybe later we should review this later (abstraction of `runner`)?

Reply:
Sorry dumb question - why must it be bound to the Trial instance?

Reply:
It seems  that the checkpoint mechanism in Ray is not well designed, and I just simply keep the original implementation. I suggest to create another issue to discuss the details.